### DISCUSSION TITLE
[PSD9] Domande Slide 18

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#164] [Esercizi (A-L)]
Domande:

- Quante porte AND servono per fare un MUX ad 8 ingressi utilizzando una rete AND/OR a 2 livelli?
- Quanti ingressi hanno le AND?
- Quanti ingressi ha l’OR che raccoglie le uscite degli AND?

**Comment by Elia-Belli** [comment]
- Abbiamo al massimo 8 minterm quindi usiamo 8 AND.
- Ogni minterm ha 1 literal per il valore + 3 literals per il segnale, quindi gli AND hanno 4 ingressi.
- L'OR ha 8 ingressi, uno per ogni minterm.

**Comment by fraxb04** [comment]
non sono d'accordo riguardo il secondo punto, poiché per un AND abbiamo 2 ingressi, ovvero il selettore che entrerà in ogni porta (o in forma literal o complement) e uno tra gli 8 input iniziali.
Inoltre, il bro non ha aggiunto le domande della slide 18, ma sono presenti solo quelle della slide 23 
(EDIT: penso di avere le slide vecchie oppure aggiornate, avendone 23, cmq mancano le domande dello XOR)

**Comment by Elia-Belli** [comment]
@fraxb04 in un $mux[8:1]$ il selettore è costituito da 3 bit, con un solo bit riusciremmo a distinguere solo 2 input diversi, con 3 bit fino a $2^3=8$

**Comment by fraxb04** [comment]
si scusami @Elia-Belli , hai ragione, quindi quando occorre individuare la quantità di bit necessari per un esercizio, prendo la quantità di output che deve essere uguale a 2 alla N (input = 2**N)?

**Comment by Elia-Belli** [comment]
Dipende dall'esercizio, però in genere per un $Mux[n:1]$ avrai sempre 1 bit dedicato al segnale ed $m$ bit decati al selettore dove $m = log_2\ n$

**Comment by fraxb04** [comment]
esatto, M è l'esponente che devo dare a 2 per ottenere N

**Comment by Amirhossein-2020** [comment]
Scusi ma non mi torna il suo raggionamento in riguardo al secondo punto, perché AND deve avere signal inputs? Se volessimo riprodurre il MUX con la logica a 2 livelli, AND non avrebbe signal input, o sbaglio?
